### PR TITLE
Add a custom Logger class with both verbose and trace levels.

### PR DESCRIPTION
**Pull request by Drapersniper**
### Description of the changes

`logging.getLogger()` returns this custom class which has the 2 new logging levels.

~~Things to do~~  
- ~~Decide how to increase the verbosity of logs via CLI, I think adding `--trace` wouldn't be a good approach. a counter for `-v` would be the idea that's the industry standard for verbosity. (`-v` would enable debug, `-v -v ` or `-vv` would enable trace)~~

CLI now accepts `-v` and `--verbose` in addition to `--debug`
CLI is no longer a toggle but a counter  - Meaning `-v` will increase the verbosity of logs to DEBUG, and `-v -v -v` will increase it to TRACE

### Have the changes in this PR been tested?

Yes

### CODE REVIEW

**Comment by Drapersniper** [comment]
Updated PR to remove dep due to its license - Added a custom Logger class instead

**Comment by Drapersniper** [comment]
image
Latest commit regex tests

**Comment by Drapersniper** [comment]
Merge first

File: redbot/setup.py [hunk 1]
```diff
@@ -352,7 +356,7 @@ async def remove_instance_interaction() -> None:
 
 
 @click.group(invoke_without_command=True)
-@click.option("--debug", type=bool)
+@click.option("--debug", "-d", "--verbose", "-v", count=True)
```
Comment (Drapersniper) [inline redbot/setup.py]:
```suggestion
@click.option("--debug", "--verbose", "-v", count=True)
```

**Review by Flame442 — DISMISSED** [review]
image
Functions as expected. I have no opinions on the colors for each of the levels, however here is a screenshot of how it currently looks if anyone has a strong opinion that it should be changed.

**Review by Flame442 — APPROVED** [review]
image
image

**Pull request merged**